labs/fabric-topologies.md

```diff
@@ -0,0 +1,40 @@
+---
+layout: default
+title: Topologies for Hyperledger Fabric
+parent: Labs
+---
+# Lab Name
+[Topologies for Hyperledger Fabric](https://github.com/hyperledger-labs/fabric-topologies)
+
+# Short Description
+Scripts and configurations to easily setup various Hyperledger Fabric topologies that exercise advanced Hyperledger Fabric capabilities. Goals:
+- facilitate the learning of the product and its capabilities
+- encourage the community to leverage these advanced capabilities when setting up production networks
+
+# Scope of Lab
+
+At [Alphaledger](https://www.alphaledger.com/) we are setting up complex multi-orgs Hyperledger Fabric networks to support the trading of municipal bonds in the US markets. Through the process we had a need to configure, implement and demonstrate in our networks some of the most advanced capabilities of the product, e.g.:
+
+ - Channel Participation API
+ - External World State Database (Couch DB Clusters)
+ - Mutual TLS for all network communications
+ - External Chaincode
+ - LDAP-enabled Fabric CAs
+ - Private Data Collections
+
+To help with our understanding of the product we have setup individual network topologies that exercised each of these cababilities. The setup of these topologies requires very few installs on a developer's machine (Docker + Docker compose) and standing up a network is simple (just the execution of a start-up script).
+
+Over time, we and the community can add more topologies to demonstrate the setup and configurations for other product features. 
+
+More background and information on the scope of this proposed lab is provided at the root documentation page for the the [pre-existing repository](https://github.com/alphaledger/hlfabric-topologies).
+
+# Initial Committers
+- [Alex Olaru](https://github.com/alexolaruhf)
+- [Taylor Trott](https://github.com/Enchilada3687)
+
+# Sponsor
+- [Andrew Boswell](dboswell@linuxfoundation.org) - Senior Director of Community Architecture, Linux Foundation
+- [David Enyeart](enyeart@us.ibm.com) - a TSC member, a maintainer of Hyperledger Fabric
+
+# Pre-existing repository
+- https://github.com/alphaledger/hlfabric-topologies
```

Comment:
It looks like the commit in this repository does not contain DCO signoff. Hyperledger requires that all contributed code has a DCO sign off. Looking at these repositories, it appears that this is not the case. There are two options that we normally consider for fixing this:

1. Amend existing commits in the repository to include DCO sign off
1. Manually transfer the code from the existing repo to the newly created repo within the hyperledger-labs

Secondly, it appears that this code is currently MIT licensed. According to [Hyperledger's charter](https://hyperledger.org/about/charter):
> Members agree that all new inbound code contributions to HLP shall be made under the Apache License, Version 2.0 (available at http://www.apache.org/licenses/LICENSE-2.0). All contributions shall be accompanied by a Developer Certificate of Origin sign-off (http://developercertificate.org/) that is submitted through a Governing Board and LF-approved contribution process. Such contribution process will include steps to also bind non-Member Contributors and, if not self-employed, their employer, to the licenses expressly granted in the Apache License, Version 2.0 with respect to such contribution.

Reply:
2 would be fine for us. I will check with our legal folks at Alphaledger to confirm that there is no issue with the APL 2.0 license and confirm soon.

Reply:
Hi Tracy,

So I would like to follow-up on all 3 open items:

1. Changing the first name for dboswell@linuxfoundation.org: I just had a call with him this morning. He told me he goes by David and not Andrew. So I think I should hold off on this change until either you or him confirm if I need to make the change?

2. Alphaledger is fine with using the APL 2.0 license.

3. We can proceed with the manual transfer of the code to the newly created repo in hyperledger-labs. Please let us know if we can help with this. Would Taylor Trott and I (the committers for this codebase) have write permissions to the new repo in hyperledger-labs? Sorry for the basic questions, we're new to this process. 

Thank you,
Alex

Reply:
> Changing the first name for [dboswell@linuxfoundation.org](mailto:dboswell@linuxfoundation.org): I just had a call with him this morning. He told me he goes by David and not Andrew. So I think I should hold off on this change until either you or him confirm if I need to make the change?

I am not sure I understand. David goes by David so why would you not change to David?

> Alphaledger is fine with using the APL 2.0 license.

Great. Please let us know when updated and we can get this lab approved and a repo created.

Reply:
Duh...sorry about that Tracy. David's correct first name is reflected now in the PR.

Thank you. Looking forward to the approval.

Reply:
I will wait until the license has been updated in the existing repo before approving. Please let me know when that has been completed.

Reply:
I pushed the commit to change the license to APL 2.0.

Reply:
> We can proceed with the manual transfer of the code to the newly created repo in hyperledger-labs. Please let us know if we can help with this. Would Taylor Trott and I (the committers for this codebase) have write permissions to the new repo in hyperledger-labs? Sorry for the basic questions, we're new to this process.

Noticed I never answered this question. Once the repo is created, you will be given permissions to write to this repo. From there, we will want you to manually migrate the code (with a PR that has DCO sign-off). Then future changes should be committed to this repo via PRs.

Reply:
Makes sense. We can do that. Thank you Tracy.